src/streams/TopicItem.js

```diff
@@ -45,20 +54,25 @@ export default function TopicItem(props: Props) {
   const backgroundData = useSelector(state => ({
     auth: getAuth(state),
     mute: getMute(state),
+    streams: getStreamsById(state),
     subscriptions: getSubscriptions(state),
+    unreadStreams: getUnreadStreams(state),
     ownUser: getOwnUser(state),
     flags: getFlags(state),
   }));
 
+  const currentStream = [...backgroundData.streams.entries()].find(x => x[1].name === streamName);
```

Comment:
I'd like to avoid introducing this sort of linear scan through all the streams.

We have a number of those now, but it's a pattern that's potentially quite bad for performance and I'd like to get rid of over time; that's #3339.

Fortunately it's not too hard to avoid: we just need an appropriate data structure like a `Map`.

If you look in `subscriptionSelectors.js` (which has selectors for both streams and subscriptions -- that's itself something we should perhaps do something to clarify), you'll see that we currently have memoizing selectors for three of the four cells in a 2x2 matrix:
```js
export const getStreamsById: Selector<Map<number, Stream>> = createSelector(
  getStreams,
  streams => new Map(streams.map(stream => [stream.stream_id, stream])),
);

export const getSubscriptionsById: Selector<Map<number, Subscription>> = createSelector(
  getSubscriptions,
  subscriptions =>
    new Map(subscriptions.map(subscription => [subscription.stream_id, subscription])),
);

export const getSubscriptionsByName: Selector<Map<string, Subscription>> = createSelector(
  getSubscriptions,
  subscriptions => new Map(subscriptions.map(subscription => [subscription.name, subscription])),
);
```

The one we need here is the remaining case: a `getStreamsByName`. So the first thing to do is to add a selector like that, following the examples of those three.

Then we can add a `streamsByName: getStreamsByName(state)` to the `backgroundData` found here and wherever else we need to turn a stream name into a `Stream` and/or a stream ID; and then we can do that by saying `streamsByName.get(streamName)`.